File: src/jv/abstractFactory/XtudoTipos/XtudoIngredientesFactory.java
package jv.abstractFactory.XtudoTipos;

import jv.abstractFactory.Ingredientes.Hamburguer;
import jv.abstractFactory.Ingredientes.Maionese;
import jv.abstractFactory.Ingredientes.Pao;

public abstract class XtudoIngredientesFactory {

    public abstract Pao criarPao();

    public abstract Hamburguer criarAmburguer();

    public abstract Maionese criarMaionese();
}

File: src/abstractFactory/xTudoTipos/XTudoIngredientesFactory.kt
package abstractFactory.xTudoTipos

import abstractFactory.ingredientes.*

/*
Created by Dhionatã on 05/02/2022
*/

abstract class XTudoIngredientesFactory {
    abstract fun criarPao(): Pao
    abstract fun criarHamburguer(): Hamburguer
    abstract fun criarMaionese(): Maionese
}